lxd/api_project.go

```diff
@@ -1244,7 +1244,25 @@ func projectValidateConfig(s *state.State, config map[string]string, defaultNetw
 		// ---
 		//  type: string
 		//  shortdesc: Cluster groups that can be targeted
-		"restricted.cluster.groups": validate.Optional(validate.IsListOf(validate.IsAny)),
+		"restricted.cluster.groups": func(value string) error {
+			if value == "" {
+				return nil
+			}
+
+			groupNames := shared.SplitNTrimSpace(value, ",", -1, true)
+			return s.DB.Cluster.Transaction(s.ShutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
```

Comment:
Rather than a new ClusterGroupsExist function, can we use `GetClusterGroups` and then test for which item(s) weren't found and either return the first one not to be found or all of the not found ones? I feel this would be more useful to the end user and avoid the need for a new query function.

Reply:
Done